Add stripping of trailing whitespace to pre-commit

suzil opened the pull request:
<!-- Please be sure to check out our contributing guidelines,
https://github.com/radis/radis/blob/develop/CONTRIBUTING.md . -->

### Description
<!-- Provide a general description of what your pull request does. -->

Add a pre-commit hook to strip trailing whitespace. It's commit noise, and this pre-commit hook will keep code consistent by stripping it always.

suzil commented:
I'm intrigued as to why CI fails from stripping whitespace 😁

erwanp commented:
The `.spec` files are how we store our Spectrum objects with metadata. They're either renamed .json, or compressed as binary. Looks like pre-commit [modified the binaries](https://github.com/radis/radis/pull/142/files?file-filters%5B%5D=.spec).

suzil commented:
Excluded those files:

```yaml
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.3.0
    hooks:
      - id: trailing-whitespace
        exclude: |
            (?x)(
                ^.+.spec|
                ^.+.dat
            )
```

erwanp commented on .pre-commit-config.yaml:
```diff
@@ -8,5 +8,14 @@ repos:
+        exclude: |
+            (?x)(
+                ^.+.spec|
+                ^.+.dat
```
Hello ! I also excluded the .levels files, which are some other plain text input files. 

```suggestion
                ^.+.dat|
                ^.+.levels
```

erwanp reviewed (changes requested):
suggested another change, otherwise we're good to go !

suzil commented:
```yaml
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.3.0
    hooks:
      - id: trailing-whitespace
        exclude: |
            (?x)(
                ^.+\.spec|
                ^.+\.dat|
                ^.+\.levels
            )
```
- Added `*.levels`
- Fixed regex to escape the period before the file extension
